lib/broccoli/default-packager.js

```diff
@@ -237,7 +237,7 @@ module.exports = class DefaultPackager {
     if (this._cachedProcessedAppAndDependencies === null) {
       let config = this.packageConfig();
       let internal = this.packageEmberCliInternalFiles();
-      let templates = this.processTemplates(allTrees);
+      let templates = this.isModuleUnificationEnabled ? null : this.processTemplates(allTrees);
```

Comment:
does this need tests?

what is the intention here?

Reply:
> does this need tests?

Probably? :D

> what is the intention here?

You can find a detailed description in #8196.